unitTests/ui/ResultListTest.ts

```diff
@@ -784,44 +784,81 @@ export function ResultListTest() {
         describe('enableScrollToTop set to', () => {
           let infiniteScrollContainer: HTMLElement;
           let allResult: IQueryResults;
-          beforeEach(() => {
-            infiniteScrollContainer = { scrollTop: 500 } as HTMLElement;
-            allResult = FakeResults.createFakeResults(50);
-          });
-
-          it('true, should change the value of ScrollTop the HTMLElement to 0', done => {
-            const option: IResultListOptions = {
+          let scrollToSpy: jasmine.Spy;
+          let scrollTopSpy: jasmine.Spy;
+          const setupResultList = (
+            option: IResultListOptions = {
               enableInfiniteScroll: true,
               infiniteScrollContainer: infiniteScrollContainer,
               enableScrollToTop: true
-            };
-            test = Mock.basicComponentSetup<ResultList>(ResultList, option);
+            }
+          ) => {
+            let test = Mock.basicComponentSetup<ResultList>(ResultList, option);
             test.cmp.currentlyDisplayedResults.push(allResult.results[1]);
             (test.cmp.queryController.fetchMore as jasmine.Spy).and.returnValue(Promise.resolve(allResult));
+            return test;
+          };
+
+          beforeEach(() => {
+            jasmine.clock().install();
+            document.createElement('div');
+            infiniteScrollContainer = document.createElement('div');
+
+            scrollToSpy = spyOn(infiniteScrollContainer, 'scrollTo');
+
+            // Inner value of scrollTop.
+            let innerValue = 500;
```

Comment:
Is this variable needed? Same for the arrow function logic on line 811?

Reply:
yes this one is needed check de 5 line after ;)